src/action.c

```diff
@@ -571,42 +575,34 @@ action_list_free(struct wl_list *action_list)
 }
 
 static void
-show_menu(struct server *server, struct view *view, const char *menu_name)
+show_menu(struct server *server, struct view *view,
+		const char *menu_name, bool at_cursor)
 {
 	if (server->input_mode != LAB_INPUT_STATE_PASSTHROUGH
 			&& server->input_mode != LAB_INPUT_STATE_MENU) {
 		/* Prevent opening a menu while resizing / moving a view */
 		return;
 	}
 
-	bool force_menu_top_left = false;
 	struct menu *menu = menu_get_by_id(server, menu_name);
 	if (!menu) {
 		return;
 	}
-	if (!strcasecmp(menu_name, "client-menu")) {
-		if (!view) {
-			return;
-		}
-		enum ssd_part_type type = ssd_at(view->ssd, server->scene,
-			server->seat.cursor->x, server->seat.cursor->y);
-		if (type == LAB_SSD_BUTTON_WINDOW_MENU) {
-			force_menu_top_left = true;
-		} else if (ssd_part_contains(LAB_SSD_PART_TITLEBAR, type)) {
-			force_menu_top_left = false;
-		} else {
-			force_menu_top_left = true;
-		}
+
+	int x = server->seat.cursor->x;
+	int y = server->seat.cursor->y;
+
+	/* The client menu needs an active client */
+	if (!view && strcasecmp(menu_name, "client-menu") == 0) {
+		return;
 	}
 
-	int x, y;
-	if (force_menu_top_left) {
+	/* Place menu in the view corner if desired (and menu is not root-menu) */
+	if (!at_cursor && strcasecmp(menu_name, "root-menu") != 0) {
 		x = view->current.x;
 		y = view->current.y;
```

Comment:
This will break for custom menu names where there is no active view.
Do we actually need the exception for the "root-menu" here? If somebody sets it up to open as some kind of custom window menu, why not.

So I'd suggest something like this instead
```c
if (!at_cursor && view) {
	x = view->current.x;
	y = view->current.x;
}
```

Reply:
Done